.buildkite/pipeline.yml

```diff
@@ -1,55 +1,94 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+---
+
 # For convenience with the many install_swiftpm_dependencies steps, we declare agents and env for macOS in the root, then specialize the fewer remaining steps
 agents:
   queue: mac
 env:
   IMAGE_ID: xcode-15.4
 
 steps:
-  - label: "🕵️ Lint"
+  - label: ":sleuth_or_spy: Lint"
     command: make lint
+    notify:
+      - github_commit_status:
+          context: Lint
     agents:
       queue: default
 
-  - label: "🔬 Test"
+  - label: ":microscope: Test"
     command: make test
+    notify:
+      - github_commit_status:
+          context: Test
     agents:
       queue: default
 
-  - label: "☢️ Danger - PR Check"
+  - label: ":radioactive_sign: Danger - PR Check"
     command: danger
     key: danger
     if: build.pull_request.id != null
     retry:
       manual:
         permit_on_passed: true
+    notify:
+      - github_commit_status:
+          context: Danger - PR Check
     agents:
       queue: linter
 
   - group: ":swift: install_swiftpm_dependencies Tests"
     steps:
       - label: ":swift: Standalone Swift Package - Autodetect"
         command: tests/install_swiftpm_dependencies/test_scripts/test_package_automatic.sh
+        notify:
+          - github_commit_status:
+              context: Standalone Swift Package - Autodetect
 
       - label: ":swift: Standalone Swift Package - Explicit"
         command: tests/install_swiftpm_dependencies/test_scripts/test_package_explicit.sh
+        notify:
+          - github_commit_status:
+              context: Standalone Swift Package - Explicit
```

Comment:
I wonder if it would make sense to prefix the `context` of all of those with `install_swiftpm_dependencies Tests: ` or `[install_swiftpm_dependencies Tests]`, so that they make more sense when seen in as list of checks in a GitHub PR?

```suggestion
              context: install_swiftpm_dependencies Tests: Standalone Swift Package - Explicit
```

```suggestion
              context: [install_swiftpm_dependencies Tests] Standalone Swift Package - Explicit
```

Reply:
Let's see how it looks: updated on 6d4444fc47fcdbbc7692c60ca5f6209658db9ff3

Reply:
Looks nicely grouped:
<img width="1049" alt="Screenshot 2024-09-10 at 21 25 47" src="https://github.com/user-attachments/assets/e39b2d98-dae5-4a26-91d5-c6cc284dd605">

Reply:
@AliSoftware @mokagio 
I've deployed https://github.com/Automattic/buildkite-ci/pull/516, updated the required checks in this repo and ran CI again.

It looks good, but there's still a check for the entire pipeline, I believe, and that one uses a name based on the Buildkite slug it seems.
I guess this is related to the setting `publish_commit_status` and shows up both in the commit history..
<img width="1061" alt="Screenshot 2024-09-11 at 12 19 24" src="https://github.com/user-attachments/assets/78c23115-7cba-4a3e-9ccc-dc94eda6921a">

...and in the PR checks.

<img width="1053" alt="Screenshot 2024-09-11 at 12 19 41" src="https://github.com/user-attachments/assets/3fcdce39-64b9-4e7f-9458-bcbeed81675c">

I think there's value in keeping it, as it seems to represent the success / failure of the entire pipeline; perhaps, for this repo, that can be the only required check for this repo as opposed to individual checks.
Having the checks on the commits is also interesting (though often intermediate commits will have their builds cancelled if a newer commit is pushed, per our settings).

Reply:
> I think there's value in keeping this setting enabled, as it seems to represent the success / failure of the entire pipeline; perhaps, for this repo, that can be the only required check for this repo as opposed to individual checks.

Sounds good to me in the context of this project. I guess that was the intention behind making it required when that setting was made (just a guess, I don't remember the early days of this repo).

> Having the checks on the commits is also interesting (though often intermediate commits will have their builds cancelled if a newer commit is pushed, per our settings).

Unless I'm missing something, that is the default behavior in GitHub for every PR where you push commits after having opened the PR. [Example](https://github.com/Automattic/simplenote-ios/pull/1655):

![image](https://github.com/user-attachments/assets/5a737c0c-eb8d-4a20-95e1-7b116196d6a6)

I agree that it's interesting. The more info, the better. I've been in a handful of situations where having those checks in the GitHub UI was useful to quickly identify which commit broke the build.

Reply:
Agreed on all counts 👍